framework/src/transfers/MultiAppVariableValueSamplePostprocessorTransfer.C

```diff
@@ -41,15 +47,104 @@ MultiAppVariableValueSamplePostprocessorTransfer::validParams()
+    _var_name(getParam<VariableName>("source_variable")),
+    _comp(getParam<unsigned int>("source_variable_component")),
+    _var(_fe_problem.getVariable(0, _var_name))
 {
   if (_directions.size() != 1)
     paramError("direction", "This transfer is only unidirectional");
 
-  if (isParamValid("from_multi_app"))
-    paramError("from_multi_app", "This transfer direction has not been implemented");
+  if (_directions.contains("from_multiapp"))
+  {
+    // Check that the variable is a CONSTANT MONOMIAL.
+    auto & fe_type = _var.feType();
+    if (fe_type.order != CONSTANT || fe_type.family != MONOMIAL)
+      paramError("source_variable",
+                 "Variable must be in CONSTANT MONOMIAL when transferring from a postprocessor "
+                 "from sub-apps.");
+
+    if (!_fe_problem.getAuxiliarySystem().hasVariable(_var_name))
+      paramError("source_variable", "Variable must be an auxiliary variable");
+  }
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::setupPostprocessorCommunication()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  const auto num_global_apps = getFromMultiApp()->numGlobalApps();
+
+  // Setup the communication pattern
+  _postprocessor_to_processor_id.resize(num_global_apps,
+                                        std::numeric_limits<processor_id_type>::max());
+  for (const auto i : make_range(num_global_apps))
+    if (getFromMultiApp()->hasLocalApp(i))
+      _postprocessor_to_processor_id[i] = this->processor_id();
+
+  _communicator.min(_postprocessor_to_processor_id);
+#ifdef DEBUG
+  for (const auto i : make_range(num_global_apps))
+  {
+    mooseAssert(_postprocessor_to_processor_id[i] != std::numeric_limits<processor_id_type>::max(),
+                "Every element in the vector should have been set.");
+    if (getFromMultiApp()->hasLocalApp(i))
+      mooseAssert(_postprocessor_to_processor_id[i] == this->processor_id(),
+                  "If I owned this app, then the processor id value should be my own");
+  }
+#endif
+}
+
+void
+MultiAppVariableValueSamplePostprocessorTransfer::cacheElemToPostprocessorData()
+{
+  if (!_directions.contains("from_multiapp"))
+    return;
+
+  // Cache the Multiapp position ID for every element.
+  auto & mesh = _fe_problem.mesh().getMesh();
+  unsigned int multiapp_pos_id = 0;
+  for (auto & elem : as_range(mesh.active_local_elements_begin(), mesh.active_local_elements_end()))
+    // Exclude the elements without dofs.
+    if (_var.hasBlocks(elem->subdomain_id()))
+    {
+      Real distance = std::numeric_limits<Real>::max();
+      unsigned int count = 0;
+      for (unsigned int j = 0; j < getFromMultiApp()->numGlobalApps(); ++j)
+      {
+        Real current_distance = (getFromMultiApp()->position(j) - elem->true_centroid()).norm();
+        if (MooseUtils::absoluteFuzzyLessThan(current_distance, distance))
+        {
+          distance = current_distance;
+          multiapp_pos_id = j;
+          count = 0;
```

Comment:
```suggestion
          distance = current_distance;
          if (elem->id() < min_elem_id)
          {
            multiapp_pos_id = j;
            min_elem_id = elem->id()
          }
          count = 0;
```

Reply:
Why do we need this `min_elem_id`?

Reply:
yeah we really need min_app_id not min_elem_id

Reply:
this implicitly uses the min app id since we are iterating over the number of apps and if distances are equal we use the first one we encounter. App ID is more deterministic than element ID ... element IDs may be affected by mesh distribution